lib/vanagon/platform.rb

```diff
@@ -271,5 +272,15 @@ def is_cross_compiled?
     def is_cross_compiled_linux?
       return (is_cross_compiled? && is_linux?)
     end
+
+    # Utility matcher to determine if the platform uses host OS provided build tools
+    # (toolchain, boost, etc).
+    #
+    # @return [true, false] whether platform uses host OS toolchain, determined by
+    # whether the platform config set the use_os_toolchain flag (defaults to true
+    # if not set)
+    def uses_os_toolchain?
+      !!@use_os_toolchain
```

Comment:
This isn't going to work the way you hope it will due to the way that booleans work. Strings are always true unless you parse them first.

```ruby
class CrapulenceFirkin
  attr_accessor :boolean

  def initialize(boolean_value)
    @boolean = boolean_value
  end

  def pure_boolean
    puts (boolean).inspect
    puts (!boolean).inspect
    puts (!!boolean).inspect
    !!boolean
  end
end
```

Behold!

```ruby
class CrapulenceFirkin
  attr_accessor :boolean

  def initialize(boolean_value)
    @boolean = boolean_value
  end

  def pure_boolean
    puts (boolean).inspect
    puts (!boolean).inspect
    puts (!!boolean).inspect
    !!boolean
  end
end
```

Here's that toy class in use:

```ruby
>> truthiness = CrapulenceFirkin.new("true")
=> #<CrapulenceFirkin:0x007fcc62e54378 @boolean="true">
>> truthiness.boolean
=> "true"
>> truthiness.pure_boolean
"true" # <= original value is a string (true)
false  # <= first inversion is a pure boolean
true   # <= second inversion is a pure boolean corresponding to
       #    the original string being considered true.
=> true
>> falsiness = CrapulenceFirkin.new("false")
=> #<CrapulenceFirkin:0x007fcc6040fd38 @boolean="false">
>> falsiness.boolean
=> "false"
>> falsiness.pure_boolean
"false" # <= original value is a string
false   # <= first inversion is a pure boolean
true    # <= second inversion is a pure boolean corresponding to
        #    the original string being considered true...
        #    but the desired value is false, not true.
=> true
>>
```

If you parse/match the value, you can get what you're looking for. Here's an example I added to that toy class above:

```ruby
  def truthiness
    return false if %w(false f no n).include? @boolean
    !!@boolean
  end
```

And in use:

```ruby
>> falsiness = CrapulenceFirkin.new("false")
=> #<CrapulenceFirkin:0x007f8e5a969c50 @boolean="false">
>> falsiness.truthiness
=> false
>> falsiness.boolean
=> "false"
```